src/game/default/g_combat.c

```diff
@@ -137,21 +137,28 @@ static int16_t G_CheckArmor(g_entity_t *ent, const vec3_t pos, const vec3_t norm
 	const int16_t quantity = ent->client->locals.inventory[ITEM_INDEX(armor)];
 
 	vec_t protection = 0.0;
+	vec_t burnrate = 1.0;
 
-	switch (armor->tag) {
+	// energy weaps cut through armor more
+	const int16_t type = (dflags & DMG_ENERGY) ? armor->tag-1 : armor->tag;
+
+	switch (type) {
 		case ARMOR_BODY:
-			protection = 1.0;
+			protection = 0.75;
+			burnrate = 1.35;
 			break;
 		case ARMOR_COMBAT:
 			protection = 0.66;
+			burnrate = 1.25;
 			break;
 		case ARMOR_JACKET:
-			protection = 0.5;
+			protection = 0.33;
+			burnrate = 1.15;
 			break;
 	}
 
 	const int16_t saved = Clamp(protection * damage, 0, quantity);
-	ent->client->locals.inventory[ITEM_INDEX(armor)] -= saved;
+	ent->client->locals.inventory[ITEM_INDEX(armor)] -= (saved / burnrate);
```

Comment:
So Q2 just removes `saved` from the appropriate inventory index. But I think, more interestingly, Q2 has an armor info struct that specifies the pickup quantity as well as the max quantity for a given armor. We don't have that distinction anymore (my bad). So in Q2, when you grab the body armor, you pickup 100 units, but you can carry 200 units total. The other armors similarly allow you to carry more than one item's worth. These armor info structs also contain the protection rates for normal as well as energy based damage.

```
typedef struct
{
	int		base_count;
	int		max_count;
	float	normal_protection;
	float	energy_protection;
	int		armor;
} gitem_armor_t;

gitem_armor_t jacketarmor_info	= { 25,  50, .30, .00, ARMOR_JACKET};
gitem_armor_t combatarmor_info	= { 50, 100, .60, .30, ARMOR_COMBAT};
gitem_armor_t bodyarmor_info	= {100, 200, .80, .60, ARMOR_BODY};
```

Something similar might work well for us: an array of `g_armor_info_t`, where the struct looks like:

```
typedef struct {
    const char *item;  // so that we can map these to the items themselves
    vec_t protection;
    vec_t expense;
    vec_t energy_protection;
    vec_t energy_expense;
    int16_t max;
} g_armor_info_t;
```

Places this would impact:

G_CheckArmor
G_PickupArmor
G_Give_f

Thoughts?